tests/utils/stl/test/config.py

```diff
@@ -0,0 +1,478 @@
+        val = self.lit_config.params.get(name, None)
+        if val is None:
+            val = getattr(self.config, name, None)
+            if val is None:
+                val = default
+        return val
+
+    def get_lit_bool(self, name, default=None, env_var=None):
+        def check_value(value, var_name):
+            if value is None:
+                return default
+            if isinstance(value, bool):
+                return value
+            if not isinstance(value, str):
+                raise TypeError('expected bool or string')
+            if value.lower() in ('1', 'true'):
+                return True
+            if value.lower() in ('', '0', 'false'):
+                return False
+            self.lit_config.fatal(
+                "parameter '{}' should be true or false".format(var_name))
+
+        conf_val = self.get_lit_conf(name)
+        if env_var is not None and env_var in os.environ and \
+                os.environ[env_var] is not None:
+            val = os.environ[env_var]
+            if conf_val is not None:
+                self.lit_config.warning(
+                    'Environment variable %s=%s is overriding explicit '
+                    '--param=%s=%s' % (env_var, val, name, conf_val))
+            return check_value(val, env_var)
+        return check_value(conf_val, name)
+
+    def configure(self):
+        self.configure_features()
+        self.configure_default_compiler()
+        self.configure_excludes()
+        self.configure_executor()
+        self.configure_expected_results()
+        self.configure_test_dirs()
+        self.configure_test_format()
+
+    def configure_test_format(self):
+        format_name = self.get_lit_conf('format_name', None)
+
+        if format_name is None:
+            format_name = 'STLTestFormat'
+
+        self.format_name = format_name
+
+    # TRANSITION: Respect passed params
+    def configure_test_dirs(self):
+        test_subdirs = list()
+
+        test_subdirs_file = self.get_lit_conf('test_subdirs_file', None)
+
+        if test_subdirs_file is not None:
+            test_subdirs_file = Path(test_subdirs_file)
+            test_subdirs += \
+                stl.test.file_parsing.parse_commented_file(test_subdirs_file)
+
+        test_subdirs += self.get_lit_conf('test_subdirs', [])
+
+        test_subdirs_root = Path(self.get_lit_conf('test_subdirs_root', ''))
+        self.lit_config.test_subdirs = \
+            list(map(lambda x: str((test_subdirs_root / x).resolve()),
+                     test_subdirs))
+
+    # TRANSITION: Don't hard-code features.
+    def configure_features(self):
+        self.config.available_features.add('long_tests')
+        self.config.available_features.add('c++2a')
+        self.config.available_features.add('msvc')
+        self.config.available_features.update(self.target_info.features)
+
+    def configure_test_src_root(self):
+        test_src_root = self.get_lit_conf('test_source_root', None)

```

Comment:
On line 118, `test_subdirs` was used in both places. Here, `'test_source_root'` is retrieved and stored in `test_src_root`. This seems inconsistent. (Note that later messages refer to `test_source_root`.)